classes/microphone.dre

```diff
@@ -147,14 +200,29 @@ SOFTWARE. -->
     gotLocalMediaStream = (localMediaStream) =>
       audioContext = new AudioContext()
       @_inputGain = audioContext.createGain()
+
       audioInput = audioContext.createMediaStreamSource(localMediaStream)
       audioInput.connect(@_inputGain)
-  
-      @_fftNode = audioContext.createAnalyser()
-      @_fftNode.fftSize = @fftsize
-      @_inputGain.connect(@_fftNode)
-      @_fftNode.smoothingTimeConstant = @fftsmoothing
-  
+
+      # FFT visualization
+      if @visualize
```

Comment:
It looks like a side affect of this conditional is that recording time will only be set if @visualize is true, because the onidle handler returns early unless @_fftNode is set, so I think you'll need to adjust the logic in onidle a bit